features/users/router.py

```diff
@@ -0,0 +1,107 @@
+"""Users feature endpoints"""
+from typing import List
+
+import fastapi
+from fastapi import APIRouter, HTTPException
+from starlette.responses import JSONResponse
+
+import features.users.exceptions
+from .input_models import RegisterUserInputModel, UpdateUserInputModel
+from .responses import UsersResponseModel
+
+from .operations import create_token, create_new_user, signin_user, get_all_users, get_user
+
+user_router = APIRouter()
+
+
+@user_router.post("/signup")
+async def signup(user: RegisterUserInputModel):
+    """
+    Sing up user
+
+    :param user:
+    :return:
+    """
+    try:
+        db_user = create_new_user(user)
+        return JSONResponse(
+            content={
+                "message": "Successfully created",
+                "user_data": {"id": db_user.id, "username": db_user.username, "email": db_user.email}
+            },
+            status_code=201
+        )
+    except features.users.exceptions.UserAlreadyExists:
+        raise HTTPException(
+            status_code=fastapi.status.HTTP_409_CONFLICT,
+            detail="User with this username or email already exists!"
+        )
+
+
+@user_router.post("/signin")
+async def signin(username: str, password: str):
+    """
+    Sing in user
+
+    :param username:
+    :param password:
+    :return:
+    """
+    try:
+        signin_user(username, password)
+        # Create jwt token
+        token, token_type = create_token(username)
+        return {"token": token, "token_type": token_type}
+    except features.users.exceptions.AccessDenied:
+        raise HTTPException(
+            status_code=fastapi.status.HTTP_403_FORBIDDEN,
+            detail="Incorrect username or password"
+        )
+
+
+@user_router.get("/all", response_model=List[UsersResponseModel])
+async def show_all_users():
+    """
+    Show all users
+
+    :return:
+    """
+    all_users = get_all_users()
```

Comment:
transform the users to a response models because returning the data class user will expose the passwords